crates/next-core/src/next_import_map.rs

```diff
@@ -236,6 +237,13 @@ pub async fn get_next_client_import_map(
         },
     );
 
+    if !*has_rewrites.await? {
+        import_map.insert_exact_alias(
+            "next/dist/shared/lib/router/utils/resolve-rewrites",
+            ImportMapping::Ignore.resolved_cell(),
+        );
+    }
```

Comment:
@sokra I imagine this would cause pretty catastrophic invalidation when `has_rewrites` changes.

Should this
- instead use an `ImportMapping::Dynamic` that only reads the cell in affected resolve requests?
- that would require making `ImportMappingReplacement::result` return an `Option<ImportMapResult>` though, to be able to not apply any mapping depending on the condition (just like resolve plugins also allow returning an option for this).

Reply:
it's ok. It's similar to changing the `resolveAlias` config. It will probably invalidate all resolving, but that's how it is. Filesystem reads would stay cached. It's better than what webpack did, which invalidates the whole cache on any next.config.js change